src/Core/Hooks/Model/InstanceManager.php

```diff
@@ -36,19 +37,26 @@ class InstanceManager implements ICanManageInstances
 
 	/** @var Dice */
 	protected $dice;
+	/** @var IManageAddons */
+	protected $addonManager;
 
-	public function __construct(Dice $dice)
+	public function __construct(Dice $dice, IManageAddons $addonManager)
 	{
-		$this->dice = $dice;
+		$this->dice         = $dice;
+		$this->addonManager = $addonManager;
+	}
+
+	public function loadAddons(): self
+	{
+		$this->addonManager->loadStrategies($this);
+		$this->addonManager->loadDecorators($this);
+
+		return $this;
 	}
 
 	/** {@inheritDoc} */
 	public function registerStrategy(string $interface, string $name, string $class, array $arguments = null): ICanManageInstances
 	{
-		if (!is_a($class, $interface, true)) {
```

Comment:
Huh?

Reply:
This check isn't useful unfortunately, because otherwise I cannot add factories for entities.. the `MonologFactory` returns a `LoggerInterface`, but it isn't this interface as an example